enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,348 @@
+conditions and may need to delete a Service to complete the change in scope.
+
+#### As a cluster administrator, I want to cancel a change to scope before it is completed
+
+Suppose the cluster administrator has changed the scope of an IngressController
+from "External" to "Internal", as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+Now the cluster administrator realizes that the scope should not be changed at
+this time, for example due to the disruption that changing the scope would
+cause.  The cluster administrator can cancel the change by changing
+`spec.endpointPublishingStrategy.loadBalancer.scope` back to its original value:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"External"}}}}'
+```
+
+#### As the provider of a managed service, I want to automate changing an IngressController's scope
+
+A managed service can initiate the operation and tell the operator to complete
+it automatically by annotating the IngressController and changing the scope,
+using an API call that is equivalent to the following commands:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+oc -n openshift-ingress-operator annotate ingresscontrollers/default ingress.operator.openshift.io/auto-delete-load-balancer=
+```
+
+The operator will begin and complete the scope-change operation, including
+deleting and recreating the service load-balancer if needed.
+
+
+### Implementation Details
+
+The operator's ingress controller reconciles IngressController objects.  When an
+IngressController is created, the controller sets the value of its
+`status.endpointPublishingStrategy.loadBalancer.scope` field to the value of the
+IngressController's `spec.endpointPublishingStrategy.loadBalancer.scope` field.
+With this enhancement, the controller checks whether these two fields differ,
+which would indicate that the user had changed the value of
+`spec.endpointPublishingStrategy.loadBalancer.scope`.  If the controller
+observes a modification to the spec field, the controller updates the status
+field.
+
+Additionally, if the operator is running on Azure or GCP, with this enhancement,
+the controller updates the annotations on the Kubernetes Service object for the
+IngressController.  Kubernetes's service controller and cloud provider
+implementation perform the necessary changes to change the service
+load-balancer's scope, and no further action is needed on the part of the user
+or the operator.
+
+If the operator is running on AWS or IBM Cloud, the controller does not update
```

Comment:
Is this the default case?  E.g. for other providers not listed, does it do this action or the previous?